R/interop-emmeans.R

```diff
@@ -30,8 +30,11 @@ recover_data.mmrm <- function(object, ...) { # nolint
   fun_call <- stats::getCall(object)
   # subject_var is excluded because it should not contain fixed effect.
   # visit_var is not excluded because emmeans can provide marginal mean
-  # by each visit.
-  model_frame <- stats::model.frame(object, include = c("visit_var", "response_var", "group_var"))
+  # by each visit if visit_var is not spatial.
+  model_frame <- stats::model.frame(
+    object,
+    include = c(if (!object$formula_parts$is_spatial) "visit_var", "response_var", "group_var")
```

Comment:
maybe break in 2 lines, or have an explicit `else NULL` ?

Reply:
updated now